pkg/volume/fc/disk_manager.go

```diff
@@ -117,5 +119,48 @@ func diskTearDown(manager diskManager, c fcDiskUnmounter, volPath string, mounte
 		}
 	}
 	return nil
+}
+
+// utility to map a device to symbolic link
+func deviceSetUp(manager diskManager, b fcDiskMapper, volPath string) error {
+	globalPDPath := manager.MakeGlobalPDName(*b.fcDisk)
+	if !filepath.IsAbs(globalPDPath) || !filepath.IsAbs(volPath) {
+		return fmt.Errorf("These paths should be absolute: globalPDPath: %s, volPath: %s", globalPDPath, volPath)
+	}
+	devicePath, err := os.Readlink(globalPDPath + "/" + "symlink")
+	if err != nil {
+		glog.Errorf("failed to readlink: %s", globalPDPath+"/"+"symlink")
+		return err
+	}
+	noMnt, err := b.mounter.IsLikelyNotMountPoint(volPath)
+	if !noMnt {
+		return fmt.Errorf("%s already mounted. This volume cant' be used as raw block volume.", volPath)
+	}
+	if err != nil && !os.IsNotExist(err) {
+		glog.Errorf("cannot validate global mount path: %s", volPath)
+		return err
+	}
+	if err = os.MkdirAll(volPath, 0750); err != nil {
+		return fmt.Errorf("Failed to mkdir %s, error", volPath)
+	}
+	if err := os.Symlink(devicePath, volPath+"/"+"symlink"); err != nil && !os.IsExist(err) {
+		return err
+	}
+	return nil
+}
+
+// utility to tear down a device
+func deviceTearDown(manager diskManager, c fcDiskUnmapper, volPath string) error {
+	fi, err := os.Lstat(volPath + "/" + "symlink")
+	if err != nil {
+		return err
+	}
+	if fi.Mode()&os.ModeSymlink == os.ModeSymlink {
+		err := os.Remove(volPath + "/" + "symlink")
```

Comment:
IMO, it should remove also the directory `volPath`. Or who deletes it and when?

Reply:
ack.